homeassistant/components/temper/sensor.py

```diff
@@ -102,3 +106,30 @@ def update(self) -> None:
                 "have changed. Attempting to reset device"
             )
             reset_devices()
+
+class HumiditySensor(SensorEntity):
+    """Representation of a Temper humidity sensor."""
+
+    _attr_device_class = SensorDeviceClass.HUMIDITY
+    _attr_native_unit_of_measurement = PERCENTAGE
+
+    def __init__(self, temper_device, name):
+        """Initialize the sensor."""
+        self.set_temper_device(temper_device)
+
+        self._attr_name = f"{name} humidity"
+        self._attr_unique_id = f"temper_{temper_device.type.name}_humidity"
+
+    def set_temper_device(self, temper_device):
+        """Assign the underlying device for this sensor."""
+        self.temper_device = temper_device
```

Comment:
This method seems quite unneeded?

Reply:
:+1: Good spot, since this is only one line I've refactored this to be part of `__init__`